core/mapserver/mapserver.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package mapserver implements the TrillianMapClient interface.
+// To start with, this will use KT's internal utilities to accomplish the same functions.
+// Later this will be replaced with a simple RPC to Trillian Maps.
+package mapserver
+
+import (
+	"crypto"
+	"fmt"
+	"log"
+
+	"github.com/google/keytransparency/core/sequenced"
+	"github.com/google/keytransparency/core/transaction"
+	"github.com/google/keytransparency/core/tree"
+	"github.com/google/trillian"
+	tcrypto "github.com/google/trillian/crypto"
+	"github.com/google/trillian/util"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+// MapServer implements TrilianMap functionality.
+type MapServer struct {
+	mapID   int64
+	tree    tree.Sparse
+	factory transaction.Factory
+	sths    sequenced.Sequenced
+	signer  *tcrypto.Signer
+	clock   util.TimeSource
+}
+
+// New returns a MapServer.
+func New(mapID int64, tree tree.Sparse, factory transaction.Factory, sths sequenced.Sequenced,
+	signer crypto.Signer, clock util.TimeSource) trillian.TrillianMapClient {
+	if _, err := tree.Commit(context.Background()); err != nil {
+		log.Printf("tree.Commit(): %v", err)
+		panic("foo")
```

Comment:
Please return the error here instead of panicing. Let the caller decides what to do with it.